### ISSUE TITLE
[tf0.9.2/master] Crash during terraform plan's refresh operation

### ISSUE THREAD

**Issue by netjunki** [hashicorp/terraform#13487]
### Terraform Version
Tested with both terraform 0.9.2 and build from master branch (07ff095755010653fa4274c11d0da853c34b7996)

### Affected Resource(s)
Multiple but the crash specifically seems to be triggered by aws_elasticache_replication_group in particular.

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
These are large and complex and can't really be run without a ton of additional infrastructure to execute them. I'll see if I can produce a simpler config to repro this specific issue.

### Panic Output
The full crash.log is over 1 million lines... so I scooped out just the stack trace.

```
2017/04/08 21:38:39 [DEBUG] plugin: terraform: panic: runtime error: index out of range
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_listener.sgp", waiting for: "aws_alb_target_group.sgp_default"
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_listener.studentdiscovery", waiting for: "aws_alb_target_group.studentdiscovery_default"
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_listener.rostering", waiting for: "aws_alb_target_group.rostering_default"
2017/04/08 21:38:39 [DEBUG] plugin: terraform:
2017/04/08 21:38:39 [DEBUG] plugin: terraform: goroutine 17852 [running]:
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_target_group.bomservice", waiting for: "aws_alb.rms"
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_listener.session", waiting for: "aws_alb_target_group.session_default"
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_listener.core", waiting for: "aws_alb_target_group.core_default"
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_listener.dashboard", waiting for: "aws_alb_target_group.dashboard_default"
2017/04/08 21:38:39 [DEBUG] dag/walk: vertex "aws_alb_target_group.prefportal", waiting for: "aws_alb.core"
2017/04/08 21:38:39 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsElasticacheReplicationGroupRead(0xc449acf260, 0x51936e0, 0xc449556780, 0x0, 0x838b700)
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /Users/balau/golang/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_elasticache_replication_group.go:226 +0xa18
2017/04/08 21:38:39 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc449e447e0, 0xc449a24230, 0x51936e0, 0xc449556780, 0xc43e98d5b0, 0xc44ba02f01, 0xa831c66d983e5152)
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /Users/balau/golang/src/github.com/hashicorp/terraform/helper/schema/resource.go:314 +0x21d
2017/04/08 21:38:39 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc449977c70, 0xc449a241e0, 0xc449a24230, 0x96ae2c0, 0x0, 0xcce3f7bf6b0a08d4)
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /Users/balau/golang/src/github.com/hashicorp/terraform/helper/schema/provider.go:267 +0x91
2017/04/08 21:38:39 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc443b4f5c0, 0xc44ab08e50, 0xc44ab09290, 0x0, 0x0)
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /Users/balau/golang/src/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017/04/08 21:38:39 [DEBUG] plugin: terraform: reflect.Value.call(0xc42001acc0, 0xc448cd8948, 0x13, 0x5a561ca, 0x4, 0xc436a52f20, 0x3, 0x3, 0x0, 0xc4496da5d0, ...)
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /usr/local/opt/go/libexec/src/reflect/value.go:434 +0x91f
2017/04/08 21:38:39 [DEBUG] plugin: terraform: reflect.Value.Call(0xc42001acc0, 0xc448cd8948, 0x13, 0xc454348720, 0x3, 0x3, 0xc436c6d200, 0x82f9240, 0xc4377aca80)
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /usr/local/opt/go/libexec/src/reflect/value.go:302 +0xa4
2017/04/08 21:38:39 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc43a7243c0, 0xc43a724380, 0xc43d438bc8, 0xc43cb71e00, 0xc45eea2840, 0x4c46820, 0xc44ab08e50, 0x16, 0x4c46860, 0xc44ab09290, ...)
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /usr/local/opt/go/libexec/src/net/rpc/server.go:387 +0x144
2017/04/08 21:38:39 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2017/04/08 21:38:39 [DEBUG] plugin: terraform:  /usr/local/opt/go/libexec/src/net/rpc/server.go:481 +0x404
```

### Expected Behavior
Terraform plan should have completed.

### Actual Behavior
Crash

### Important Factoids
Was seeing unexpected EOF/connection is shut down errors during prior runs before bringing this over to my laptop to investigate where I encountered the crash.

Seeing hundreds of errors from these running from a linux host:
```
* module.transform_passthrough_stack.module.groupBridgeTransformation-elasticache.aws_elasticache_subnet_group.subnet_group: aws_elasticache_subnet_group.subnet_group: connection is shut down
* aws_alb_target_group.goalmanagementportal: aws_alb_target_group.goalmanagementportal: unexpected EOF
* module.redis_growth.aws_elasticache_replication_group.elasticache: aws_elasticache_replication_group.elasticache: connection is shut down

```

**Comment by netjunki** [comment]
Took a stab at a fix #13488

**Comment by netjunki** [comment]
Once running my templates with the fix I made in place I encountered a collection of "Not enough IP space available in subnet-b95b17cf" and I'm now curious if the initial error of the NodeGroupMembers being empty might have be caused by a lack of available IPs in the first place.

**Comment by catsby** [comment]
Fixed the crash in #13488 , thanks!
